database/migrations/2014_10_12_000000_create_users_table.php

```diff
@@ -17,8 +17,8 @@ public function up()
             $table->id();
             $table->string('first_name', 100);
             $table->string('surname', 100);
-            $table->string('email', 250);
-            $table->string('hash', 10);
+            $table->string('email', 250)->unique();
+            $table->string('hash', 10)->nullable();
```

Comment:
Why nullable?
Thought this is required.

Reply:
The hash is only required to send an email to the user. For the creation it isn't important.
Until now we don't have proper hash function. If the hash-row isn't nullable, we have to give a User a random value or a standard value like 0 or will get an error. So there isn't a big difference, if we check for null or 0. 
In my next task I will create the hash function, so we can set a hash by creation. In future the hash also must be unique.